src/js/backup_restore.js

```diff
@@ -321,25 +321,24 @@ function configuration_restore(callback) {
                 if ((e.total != 0 && e.total == e.loaded) || GUI.isCordova()) {
                     // Cordova: Ignore verification : seem to have a bug with progressEvent returned
                     console.log('Read SUCCESSFUL');
-
+                    let configuration;
                     try { // check if string provided is a valid JSON
-                        var configuration = JSON.parse(e.target.result);
+                        configuration = JSON.parse(e.target.result);
                     } catch (e) {
                         // data provided != valid json object
                         console.log('Data provided != valid JSON string, restore aborted.');
 
                         return;
                     }
 
-
                     // validate
-                    if (typeof configuration.generatedBy !== 'undefined' && compareVersions(configuration.generatedBy, CONFIGURATOR.BACKUP_FILE_VERSION_MIN_SUPPORTED)) {
+                    if (typeof configuration?.generatedBy && compareVersions(configuration.generatedBy, CONFIGURATOR.BACKUP_FILE_VERSION_MIN_SUPPORTED)) {
```

Comment:
`typeof configuration?.generatedBy` is always true, as `typeof x` returns a non-empty string.
`if (configuration.generatedBy !== undefined && compareVersions(configuration.generatedBy, CONFIGURATOR.BACKUP_FILE_VERSION_MIN_SUPPORTED)) {` is how I would put this, but the way it is is not actually wrong.

Reply:
Good spot! Changed (not committed as I am working on power.js 90% ready)